hollow/src/main/java/com/netflix/hollow/api/consumer/metrics/AbstractRefreshMetricsListener.java

```diff
@@ -118,6 +124,21 @@ private final void noFailRefreshEndMetricsReporting(ConsumerRefreshMetrics refre
         }
     }
 
+    @Override
+    public void announcementDetected(long newVersion, Map<String, String> metadata, boolean isPinned) {
+        /** Track the timestamp for HEADER_TAG_METRIC_ANNOUNCEMENT only when the namespace is not pinned.
+        * Store the pinned status of previous cycle to detect the scenario where a version got unpinned in the next cycle
+        * then don't record this metric as this will record the duration of current version from the last announced version.
+        * namespacePinnedPreviously == true && isPinned == true then dont record
+        * namespacePinnedPreviously == true && isPinned == false then dont record
+        * namespacePinnedPreviously == false && isPinned == true then dont record
+        * namespacePinnedPreviously == false && isPinned == false then record */
+        if (!(namespacePinnedPreviously || isPinned)) {
```

Comment:
consider slightly refactoring to- always record version to announce time in this map, but only report metric if the pin/unpin criteria is validated